server/src/logic/utils/model.utils.ts

```diff
@@ -0,0 +1,19 @@
+import { ApiError, ERRORS } from "./../../models/index";
+
+export class ModelUtils {
```

Comment:
I thought this is not really a middleware function because middleware functions typically take a res or / and a req object and are a part of the request chain.

This function should only be a simple helper. What's the definition of a middleware function?